UWPInjector/source/main.cpp

```diff
@@ -211,14 +215,69 @@ int main(int argc, char** argv, char** envp)
 
 	std::cout << "Remote Dumper thread found: 0x" << std::hex << IPC::GetTargetThread() << std::endl;
 
+	//get time for filename
+	time_t result = time(NULL);
+	char str[26];
+	ctime_s(str, sizeof str, &result);
+	std::string LogFileName = str;
+	std::replace(LogFileName.begin(), LogFileName.end(), ':', '-');
+	LogFileName.erase(std::remove(LogFileName.begin(), LogFileName.end(), '\n'), LogFileName.end());
+
+	//get proccess name for filename
+	HANDLE hProcess = OpenProcess(PROCESS_ALL_ACCESS | PROCESS_QUERY_INFORMATION |
+		PROCESS_VM_READ,
+		FALSE, ProcessID);
+
+
+	if (hProcess)
+	{
+		TCHAR Buffer[MAX_PATH];
+		if (GetModuleFileNameEx(hProcess, 0, Buffer, MAX_PATH))
+		{
+			// At this point, buffer contains the full path to the executable
+			std::filesystem::path FileName(Buffer);
+			std::string tempstring = FileName.stem().generic_string();
+			LogFileName = FileName.stem().generic_string() + " " + LogFileName;
+		}
+		else
+		{
+			std::cout << "\033[91mFailed to get process Name" << std::endl;
+			system("pause");
+			return EXIT_FAILURE;
+		}
+		CloseHandle(hProcess);
+	}
+	
+	//attempt to create file
+	std::cout << LogFileName;
+	std::wofstream LogFile((LogFileName + ".txt"));
+	bool writing;
+	if (LogFile.is_open())
+	{
+		std::cout << "\033[92mLogging to File: " << LogFileName << ".txt"<<"\n\033[0m";
+		writing = true;
+	}
+	else
+	{
+		std::cout << "\033[91mFailed to open log file\n\033[0m";
+		writing = true;
+	}
+
 	std::cout << "\033[0m" << std::flush;
 	while( IPC::GetTargetThread() != IPC::InvalidThread )
 	{
 		while( IPC::MessageCount() > 0 )
 		{
-			std::wcout << IPC::PopMessage() << "\033[0m";
+			std::wstring TempMessage = IPC::PopMessage();
+			std::wcout << TempMessage << "\033[0m";
+			if (writing)
```

Comment:
See previous review involving `writing`. We might not need this variable at all if we decide to exit from failing to open the log file for writing.